### PR TITLE
made Transfer check if ready to shoot🦟™️

### LINKED ISSUE #75: Lock feeder to only feed at goal RPM and angles
Only allow shooting when at goal angle and rpm

### PR DESCRIPTION
**Pull request by BrodyKarr**

### CODE REVIEW

File: src/main/java/frc/robot/commands/Shoot.java [hunk 1]
```diff
@@ -7,16 +7,25 @@
...
+  Transfer subTransfer;
+  Turret subTurret;
+  Pitch subPitch;
 
-  public Shoot(Shooter subShooter, LEDs subLEDs) {
+  public Shoot(Shooter subShooter, LEDs subLEDs, Transfer subTransfer, Pitch subPitch, Turret subTurret) {
     this.subShooter = subShooter;
     this.subLEDs = subLEDs;
+    this.subTransfer = subTransfer;
+    this.subTurret = subTurret;
```
Comment (TaylerUva) [inline src/main/java/frc/robot/commands/Shoot.java]: Remove

**Review by TaylerUva — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/TransferGamePiece.java [hunk 2]
```diff
@@ -28,6 +43,22 @@ public void initialize() {
   public void execute() {
     globalTransfer.setFeederMotorSpeed(prefTransfer.feederMotorSpeed.getValue());
     globalTransfer.setTransferMotorSpeed(prefTransfer.transferMotorSpeed.getValue());
```
Comment (TaylerUva) [inline src/main/java/frc/robot/commands/TransferGamePiece.java]: These will always run

**Review by TaylerUva — DISMISSED** [review]

**Comment by alicekuznetsov** [comment]
Closes #167 
Closes #173

**Comment by alicekuznetsov** [comment]
@TaylerUva  do we want this in main or do we want to wait until we fix the absolute encoders? Also, what's our plan from there? Vision? I just want to have a clear picture of how Tuesday should go. Unfortunately, I think we only have one meeting left for anything software-related so we really gotta make it count 🤞. I _maybe_ could come in early on Tuesday (I have math 4 as my 5th period... i lowkey don't want to sit in there "learning" about e for 2.5 hours)

Stuff we need to do:
- Abs Encoders
- Vision testing
- Auto (tuning, testing)
- Amp preset a bit better
- Trap attempted 
- Subwoofer preset a bit better
- Further testing with auto-aim (once Vision is working)
- Driver snapping to source and chain (once Vision is working, not as bad as it sounds)
- Decide if we are keeping flywheels on all match or what

I'm thinking
Fix Abs Encoders -> Vision testing -> Subwoofer -> Amp -> Driver snapping -> Trap (very temporarily) -> flywheels??? -> Auto for the rest of our time

I really think i am forgetting something

**Comment by TaylerUva** [comment]
Let's merge this in and fix in main.

My suggestion:

1. Fix Abs Encoders (must do)
2. Vision testing (must do)
3. Trap functionality (1 hour)
4. Auto functionality (let's get a least one working)
5. Subwoofer (good enough already)
6. Amp (good enough already)
7. Driver snapping (🤷)

File: src/main/java/frc/robot/subsystems/Transfer.java [hunk 3]
```diff
@@ -36,16 +41,39 @@ public Transfer() {
...
+
+    SmartDashboard.putNumber("PAIN: current > belowCurrent val", (current - lastCurrent));
+    SmartDashboard.putNumber("PAIN: velocity val",
+        (curVelocity - lastVelocity));
+
+    SmartDashboard.putNumber("AA: Transfer Current", current);
+    SmartDashboard.putNumber("AA: Transfer Vel", curVelocity);
+    SmartDashboard.putNumber("AA: Feeder Current", feederMotor.getStatorCurrent());
+
+    if ((feederMotor.getStatorCurrent() < -10) && (curVelocity < 0.8) && (current > 6) || hasGamePiece) {
```
Comment (TaylerUva) [inline src/main/java/frc/robot/subsystems/Transfer.java:64]: These need to be made into preferences (in another PR)

**Review by TaylerUva — APPROVED** [review]

**Comment by TaylerUva** [comment]
Good to merge @ACat701? I approve

**Pull request merged**